SSA/Experimental/IntrinsicAsymptotics.lean

```diff
@@ -93,29 +93,103 @@ mk_ex 120
 -- verified transformations to larger programs parsed at run time.
 
 /-- An untyped expression as an intermediate step of input processing. -/
+
+abbrev Var := Nat
 inductive Expr : Type
-  | add (a : Nat) (b : Nat)
+  | add (a : Var) (b : Var)
   | nat (n : Nat)
-  deriving Repr
+  deriving Repr, Inhabited
+
+inductive ExprRec : Type
+  | add (a : ExprRec) (b : ExprRec)
+  | nat (n : Nat)
+  | var (idx : Nat)
+  deriving Repr, Inhabited
```

Comment:
We can unify `ExprRec` and `Expr` into a single `ExprF` and a fixpoint of the initial algebra of `ExprF`. Whether this is useful is  a different question :)